forge-ai/src/main/java/forge/ai/ComputerUtilMana.java

```diff
@@ -1521,7 +1523,17 @@ public boolean apply(final Card c) {
                 final Cost cost = m.getPayCosts();
 
                 if (cost != null) {
-                    needsLimitedResources |= !cost.isReusuableResource();
+                    if (!cost.isReusuableResource()) {
+                        for(CostPart part : cost.getCostParts()) {
+                            if (part instanceof CostSacrifice) {
+                                if (!part.getType().equals("CARDNAME")) {
+                                    unpreferredCost = true;
+                                } else {
+                                    needsLimitedResources = true;
```

Comment:
it seems the logic got "degraded" a bit here:
now `needsLimitedResources` can only be true if there's a Sac Cost which seems wrong

Reply:
Ah yea, Lemme refactor a bit to get what I need.